docs/labs/flockercontainers.rst

```diff
@@ -0,0 +1,97 @@
+.. _flocker-containers:
+
+=============================
+Running Flocker in Containers
+=============================
+
+For some environments, such as CoreOS, and ECS you must install Flocker inside a container.
+
+.. warning::
+   Installing Flocker in containers is useful for environments such as CoreOS, or Amazon ECS, but please note that doing so is currently experimental.
+
+You can find the relevant Docker Hub images here:
+
+* `Dataset Agent <https://hub.docker.com/r/clusterhq/flocker-dataset-agent/>`_
+* `Container Agent <https://hub.docker.com/r/clusterhq/flocker-container-agent/>`_
+* `Control Service <https://hub.docker.com/r/clusterhq/flocker-control-service/>`_
+* `Docker Plugin <https://hub.docker.com/r/clusterhq/flocker-docker-plugin/>`_
+
+Before you install Flocker in containers, all authentication must be completed.
+This means that you will have generated the cluster certificate and key, the  node certificates and keys, and the API certificates and keys.
+
+In the example commands below, certificates are assumed to live in :file:`/etc/flocker` but other environments may choose to create containers that hold certificates and configuration, and share them amongst the Flocker service containers.
+The latter is not documented.
```

Comment:
I see that you've added this in response to @wallrj's experience on GiantSwarm. I think it's overall most clear to avoid mentioning the specifics of GiantSwarm and just say what we're assuming in the docs. e.g.:

"""
In the example commands below, certificates are assumed to live in :file:`/etc/flocker`.
"""